include/circt/Dialect/ESI/ESIOps.td

```diff
@@ -41,7 +41,23 @@ def ChannelBuffer : ESI_Op<"buffer", [NoSideEffect, SameTypeOperands]> {
   let parser = [{ return ::parse$cppClass(parser, result); }];
 }
 
-def CosimEndpoint : ESI_Op<"cosim", []> {
+def PipelineStage : ESI_Physical_Op<"stage", [NoSideEffect, SameTypeOperands]> {
+  let summary = "An elastic buffer stage.";
+  let description = [{
+    An individual elastic pipeline register. Generally lowered to from a
+    ChannelBuffer ('buffer'), though can be inserted anywhere to add an
+    additional pipeline stage. Adding individually could be useful for
+    late-pass latency balancing.
+  }];
+
+  let arguments = (ins ChannelType:$input);
+  let results = (outs ChannelType:$output);
+
+  let printer = [{ return ::print(p, *this); }];
+  let parser = [{ return ::parse$cppClass(parser, result); }];
```

Comment:
Is the C++ parser really needed here? Seems the syntax could be expressed in the assembly format specification, but I could be missing something.

Reply:
Yeah, that's what I was hoping. The declarative format requires that you specify both the input and output types, which I wanted to avoid.

Reply:
Ah. Is that still the case if you use the `SameOperandsAndResultType` trait?

Reply:
That's what I meant to do.

Reply:
And yep. I don't think the assembly format is sensitive to traits (yet).

Reply:
:+1: makes sense, I guess that could be a nice enhancement.